static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -634,6 +637,75 @@ function WidgetViewerModal(props: Props) {
               );
             }}
           </IssueWidgetQueries>
+        ) : widget.widgetType === WidgetType.METRICS ? (
+          <MetricsWidgetQueries
+            api={api}
+            organization={organization}
+            widget={tableWidget}
+            selection={modalSelection}
+            limit={
+              widget.displayType === DisplayType.TABLE
+                ? FULL_TABLE_ITEM_LIMIT
+                : HALF_TABLE_ITEM_LIMIT
+            }
+            includeAllArgs
+            cursor={cursor}
+          >
+            {({tableResults, loading, pageLinks}) => {
+              const links = parseLinkHeader(pageLinks ?? null);
+              const isFirstPage = links.previous?.results === false;
+              return (
+                <React.Fragment>
+                  <GridEditable
+                    isLoading={loading}
+                    data={tableResults?.[0]?.data ?? []}
+                    columnOrder={columnOrder}
+                    columnSortBy={columnSortBy}
+                    grid={{
+                      renderHeadCell: renderMetricsGridHeaderCell({
+                        ...props,
+                        widget: tableWidget,
+                        tableData: tableResults?.[0],
+                        onHeaderClick: () => setChartUnmodified(false),
+                      }) as (
+                        column: GridColumnOrder,
+                        columnIndex: number
+                      ) => React.ReactNode,
+                      renderBodyCell: renderGridBodyCell({
+                        ...props,
+                        tableData: tableResults?.[0],
+                        isFirstPage,
+                      }),
+                      onResizeColumn,
+                    }}
+                    location={location}
+                  />
+                  {(links?.previous?.results || links?.next?.results) && (
+                    <Pagination
+                      pageLinks={pageLinks}
+                      onCursor={newCursor => {
+                        router.replace({
+                          pathname: location.pathname,
+                          query: {
+                            ...location.query,
+                            [WidgetViewerQueryField.CURSOR]: newCursor,
+                          },
+                        });
+                        trackAdvancedAnalyticsEvent(
+                          'dashboards_views.widget_viewer.paginate',
+                          {
+                            organization,
+                            widget_type: widget.widgetType ?? WidgetType.DISCOVER,
```

Comment:
Should we plug `WidgetType.METRICS` here because the only way we get to this line is if we pass `widget.widgetType === WidgetType.METRICS` on 640